app/src/main/java/com/example/geoquiz/MainActivity.kt

```diff
@@ -54,46 +65,52 @@ class MainActivity : AppCompatActivity() {
             }
             updateQuestion()
         }
+        previousButton.setOnClickListener {
+            if (currentIndex > 0) {
+                currentIndex = (currentIndex - 1)
+            }
+            updateQuestion()
+        }
         updateQuestion()
     }
 
     private fun updateQuestion() {
         val questionTextResId = questionBank[currentIndex].textResId
         questionTextView.setText(questionTextResId)
+        answeredQuestionCheck()
     }
 
     private fun checkAnswer(userAnswer: Boolean) {
         val answer = questionBank[currentIndex].answer
-        val messageResId = if (answer == userAnswer) {
-            R.string.correct
+        var messageResId = 0
+        if (answer == userAnswer) {
+            messageResId = R.string.correct
+            correctAnswers++
         } else {
-            R.string.incorrect
+            messageResId = R.string.incorrect
         }
         Toast.makeText(this, messageResId, Toast.LENGTH_SHORT).show()
     }
 
-    override fun onStart() {
-        super.onStart()
-        Log.d(TAG, "onStart called ")
-    }
 
-    override fun onResume() {
-        super.onResume()
-        Log.d(TAG, "onResume called ")
-    }
-
-    override fun onPause() {
-        super.onPause()
-        Log.d(TAG, "onPause called ")
-    }
-
-    override fun onStop() {
-        super.onStop()
-        Log.d(TAG, "onStop called ")
+    private fun answeredQuestionCheck() {
+        val checkAnswer = questionBank[currentIndex].answeredQuestion
+        if (!checkAnswer) {
+            trueButton.isEnabled = true
+            falseButton.isEnabled = true
+            nextButton.isEnabled = false
+        } else {
+            nextButton.isEnabled = true
+            trueButton.isEnabled = false
+            falseButton.isEnabled = false
+        }
     }
 
-    override fun onDestroy() {
-        super.onDestroy()
-        Log.d(TAG, "onDestroy called ")
+    private fun getCorrectAnswerPercentage() {
```

Comment:
Good name, but all functions should have a single responsibility and all methods named get should return something.
This one computes the percentage and shows the toast. So it is not single responsibility.

Remove the toast from here, make it return the percentage and show the toast in the click listener